ui/src/app/views/cohort-edit/component.spec.ts

```diff
@@ -0,0 +1,267 @@
+      } else {
+        return false;
+      }
+    });
+  }
+
+  public getCohort(wsNamespace: string,
+                   wsId: string,
+                   cId: string): Observable<Cohort> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspace(wsNamespace, wsId);
+        if (workspaceMatch !== null) {
+          observer.next(workspaceMatch.cohorts.find(function(cohort: Cohort) {
+            if (cohort.id === cId) {
+              return true;
+            }
+          }));
+          observer.complete();
+        }
+        observer.error(new Error('Error fetching a cohort. No cohort with \
+          id: ' + cId + ' exists in workspace ' + wsNamespace + ', ' + wsId +
+          ' in cohort service stub.'));
+      }, 0);
+    });
+    return observable;
+  }
+
+  public updateCohort(wsNamespace: string,
+                      wsId: string,
+                      cId: string,
+                      newCohort: Cohort): Observable<Cohort> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspace(wsNamespace, wsId);
+        if (workspaceMatch !== null) {
+          const index = workspaceMatch.cohorts.findIndex(function(cohort: Cohort) {
+            if (cohort.id === cId) {
+              return true;
+            }
+            return false;
+          });
+          if (index != null) {
+            workspaceMatch.cohorts[index] = newCohort;
+          }
+          observer.complete();
+        }
+        observer.error(new Error('Error updating a cohort. No cohort with id: '
+          + cId + ' exists in workspace ' + wsNamespace + ', ' + wsId +
+          ' in cohort service stub.'));
+      }, 0);
+    });
+    return observable;
+  }
+
+  public createCohort(wsNamespace: string,
+                      wsId: string, newCohort: Cohort): Observable<Cohort> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspace(wsNamespace, wsId);
+        if (workspaceMatch !== null) {
+          observer.next(workspaceMatch.cohorts.find(function(cohort: Cohort) {
+            if (cohort.id === newCohort.id) {
+              observer.error(new Error('Error creating a cohort. Cohort with \
+                id: ' + cohort.id + ' already exists.'));
+              return true;
+            }
+          }));
+          workspaceMatch.cohorts.push(newCohort);
+          observer.complete();
+        }
+        observer.error(new Error('Error creating a cohort. No workspace '
+          + wsNamespace + ', ' + wsId + ' in cohort service stub.'));
+      }, 0);
+    });
+    return observable;
+  }
+
+  public getCohortsInWorkspace(wsNamespace: string,
+                               wsId: string): Observable<CohortListResponse> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
```

Comment:
We don't actually need it, however it closer replicates what the actual service would do. (Otherwise the observer completes instantaneously on creation, which is not actually what we want)